network_test.go

```diff
@@ -0,0 +1,115 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"net"
+	"reflect"
+	"testing"
+
+	pb "github.com/kata-containers/agent/protocols/grpc"
+	"github.com/stretchr/testify/assert"
+	"github.com/vishvananda/netlink"
+)
+
+func TestUpdateRemoveInterface(t *testing.T) {
+
+	s := sandbox{}
+
+	ifc := pb.Interface{
+		Name:   "enoNumber",
+		Mtu:    1500,
+		HwAddr: "02:00:ca:fe:00:48",
+	}
+	ip := pb.IpAddress{
+		Family:  0,
+		Address: "192.168.0.101",
+		Mask:    "24",
+	}
+	ifc.IpAddresses = append(ifc.IpAddresses, &ip)
+
+	netHandle, _ := netlink.NewHandle()
+	defer netHandle.Delete()
+
+	//
+	// Initial test: try to update a device which doens't exist:
+	//
+	_, err := s.updateInterface(netHandle, &ifc)
+	assert.NotNil(t, err, "Expected to observe interface couldn't be found error")
+
+	// create a dummy link that we can test update on
+	macAddr := net.HardwareAddr{0x02, 0x00, 0xCA, 0xFE, 0x00, 0x48}
+	link := &netlink.Dummy{
+		LinkAttrs: netlink.LinkAttrs{
+			MTU:          1500,
+			TxQLen:       -1,
+			Name:         "ifc-name",
+			HardwareAddr: macAddr,
+		},
+	}
+	netHandle.LinkAdd(link)
+	netHandle.LinkSetUp(link)
+
+	//
+	// With a link populated, check to see if we can successfully update:
+	//
+	resultingIfc, err := s.updateInterface(netHandle, &ifc)
+	assert.Nil(t, err, "Unexpected update interface failure: %v", err)
+	assert.True(t, reflect.DeepEqual(resultingIfc, &ifc),
+		"Interface created didn't match: got %+v, expecting %+v", resultingIfc, ifc)
+
+	// Try with a different valid MTU.  Make sure we can assign a new set of IP addresses
+	ifc.Mtu = 500
+	ifc.IpAddresses[0].Address = "192.168.0.102"
+	ip2 := pb.IpAddress{0, "182.168.0.103", "24"}
+	ifc.IpAddresses = append(ifc.IpAddresses, &ip2)
+
+	resultingIfc, err = s.updateInterface(netHandle, &ifc)
+	assert.Nil(t, err, "Unexpected update interface failure: %v", err)
+	assert.True(t, reflect.DeepEqual(resultingIfc, &ifc),
+		"Interface created didn't match: got %+v, expecting %+v", resultingIfc, ifc)
+
+	// Try with garbage:
+	//
+	ifc.Mtu = 999999999999
+	resultingIfc, err = s.updateInterface(netHandle, &ifc)
+	// expecting this failed
+	assert.NotNil(t, err, "Expected failure")
+
+	ifc.Mtu = 500
+	assert.True(t, reflect.DeepEqual(resultingIfc, &ifc),
+		"Resulting inteface should have been unchanged: got %+v, expecting %+v", resultingIfc, ifc)
+	//	assert.False(t, resultingIfc.Mtu != 500,
+	//		"Resulting interface should have been unchanged: got MTU:  %d, expecting MTU: %d", resultingIfc.Mtu, 500)
```

Comment:
Small nit: need to clean unused annotations.

This can be updated in following PRs

Reply:
Ah, should've done commit -a -s :).  Good catch